modules/distributor/distributor.go

```diff
@@ -397,15 +400,37 @@ func (d *Distributor) sendToIngestersViaBytes(ctx context.Context, userID string
 			return err
 		}
 
-		_, err = c.(tempopb.PusherClient).PushBytesV2(localCtx, &req)
+		response, err := c.(tempopb.PusherClient).PushBytesV2(localCtx, &req)
 		metricIngesterAppends.WithLabelValues(ingester.Addr).Inc()
-		if err != nil {
+
+		if err != nil { // internal error, drop entire batch
 			metricIngesterAppendFailures.WithLabelValues(ingester.Addr).Inc()
+			return err
 		}
-		return err
+
+		mu.Lock()
+		defer mu.Unlock()
+		responses = append(responses, response)
+		return nil
 	}, func() {})
+	// if err != nil, we discarded everything because of an internal error
+	if err != nil {
+		return err
```

Comment:
nit: it might be clearer at this point to move:
```
overrides.RecordDiscardedSpans(spanCount, reasonInternalError, userID)
```
in here since all discarded span counts are recorded right here.